File: word_rank.py
""" Write a program that gets from input a sentence, counts word frequencies, and prints one line for each
word, in descending order of frequency """

# Dictionary for the words
word_rank = dict()


# Function that populates the dictionary like this:
# {word:how many times the words appeared in the sentence}
def count_word(word, sentence):
    word_rank[word] = sentence.count(word)


# The sentence
text = input()
# Split the sentence in a list of words
words_in_text = text.split()

# For each word in the sentence apply the count_word function
for word in words_in_text:
    count_word(word, text)

# Create a dictionary that has the keys,sorted in reverse, of the word_rank dictionary
sorted_keys = sorted(word_rank.items(), key=lambda kv: kv[1], reverse=True)
# Iterate through the sorted_keys dictionary and print word rank: the word_rank
for key in sorted_keys:
    print(key[0], " rank:", key[1])

File: word_rank.java
/*
Write a program that gets from input a sentence, counts word frequencies, and prints one line for each word and frequencies
*/
import java.util.*;

class Solution {
	public static void count_word(String word,String[] sentence,Map<String, Integer> word_rank)
	{
		int nr_app=0;
		for (int i=0;i<sentence.length;i++)
		{
			if (sentence[i].equals(word))
			{
				nr_app+=1;
			}
		}
		word_rank.put(word,nr_app);
	}

	public static void main(String[] args) {
		Scanner scanner = new Scanner(System.in);
		System.out.println("Sentence=");
		String sentence=scanner.nextLine();
		Map<String, Integer> word_rank = new HashMap<String, Integer>();
		String[] words_in_text=sentence.split("\\s+");
		for (int i=0;i<words_in_text.length;i++)
		{
			count_word(words_in_text[i],words_in_text,word_rank);
		}
		for(String key : word_rank.keySet()) 
		{
			  System.out.println(key+":"+word_rank.get(key));
		}
	}
}
